desc/profiles.py

```diff
@@ -810,23 +810,28 @@ class SplineProfile(_Profile):
         - `'catmull-rom'`: C1 cubic centripetal "tension" splines
     name : str
         name of the profile
+    df : array-like
+        Optional. Values of the function derivative at knot locations.
 
     """
 
     _io_attrs_ = _Profile._io_attrs_ + ["_knots", "_method"]
 
-    def __init__(self, values=None, knots=None, method="cubic2", name=""):
+    def __init__(
+        self, values=None, knots=None, method="cubic2", name="", df=None, jnp=np
+    ):
         super().__init__(name)
 
         if values is None:
             values = [0, 0, 0]
-        values = np.atleast_1d(values)
+        values = jnp.atleast_1d(values)
```

Comment:
[Following up on this comment](https://github.com/PlasmaControl/DESC/pull/1003/files#r1638527983), if you keep np here instead of jnp, the error for `pytest -k test_ad_ripple tests/test_neoclassical.py` is a `list index out of range error` which looks identical to #1040 .

```
>       g = obj.grad(obj.x())

tests/test_neoclassical.py:84: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
desc/objectives/objective_funs.py:415: in grad
    return jnp.atleast_1d(self._grad(x, constants).squeeze())
desc/derivatives.py:89: in __call__
    return self.compute(*args, **kwargs)
desc/derivatives.py:150: in compute
    return self._compute(*args, **kwargs)
desc/objectives/objective_funs.py:335: in compute_scalar
    f = jnp.sum(self.compute_scaled_error(x, constants=constants) ** 2) / 2
desc/objectives/objective_funs.py:312: in compute_scaled_error
    [
desc/objectives/objective_funs.py:313: in <listcomp>
    obj.compute_scaled_error(*par, constants=const)
desc/objectives/objective_funs.py:952: in compute_scaled_error
    f = self.compute(*args, **kwargs)
desc/objectives/_neoclassical.py:235: in compute
    iota=SplineProfile(iota, df=iota_r, knots=self._rho, name="iota", jnp=jnp),
desc/profiles.py:827: in __init__
    values = np.atleast_1d(values)
../../../miniconda3/envs/desc-env/lib/python3.10/site-packages/numpy/core/shape_base.py:65: in atleast_1d
    ary = asanyarray(ary)
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 

self = TracerArrayConversionError(Traced<ShapedArray(float64[2])>with<DynamicJaxprTrace(level=5/0)>)
tracer = Traced<ShapedArray(float64[2])>with<DynamicJaxprTrace(level=5/0)>

    def __init__(self, tracer: core.Tracer):
      super().__init__(
          "The numpy.ndarray conversion method __array__() was called on "
>         f"{tracer._error_repr()}{tracer._origin_msg()}")
E     IndexError: list index out of range

```